cpp/pagerank_module/algorithm_online/pagerank.hpp

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include <random>
+#include <unordered_set>
+
+namespace pagerank_online_alg {
+namespace {
+class PageRankData {
+  ///
+  ///@brief Context for storing data for dynamic pagerank
+  ///
+  ///
+ public:
+  void Init() {
+    walks.clear();
+    walks_counter.clear();
+    walks_table.clear();
+
+    gen = new std::mt19937(42);
+    distr = new std::uniform_real_distribution<float>(0.0, 1.0);
```

Comment:
When do you delete?
That's why you need to prefer `std::unique_ptr`.

Reply:
I think actually it is not necessary. It should be possible initialize them with [default member initialization](https://en.cppreference.com/w/cpp/language/data_members#Member_initialization):
```cpp
class PageRankData {
  /// ....
  std::mt19937 gen{std::random_device{}()};
  std::uniform_real_distribution<float> distr{0.0, 1.0};
};
```

Reply:
Oh yes, I don't think currently they need to be reset in `Init`, because we don't gain any value by doing so. The "randomness" won't change when we reset them.